addons/mrp/models/mrp_unbuild.py

```diff
@@ -188,23 +188,24 @@ def action_unbuild(self):
         # TODO: Will fail if user do more than one unbuild with lot on the same MO. Need to check what other unbuild has aready took
         qty_already_used = defaultdict(float)
         for move in produce_moves | consume_moves:
-            if move.has_tracking != 'none':
-                original_move = move in produce_moves and self.mo_id.move_raw_ids or self.mo_id.move_finished_ids
-                original_move = original_move.filtered(lambda m: m.product_id == move.product_id)
-                needed_quantity = move.product_uom_qty
-                moves_lines = original_move.mapped('move_line_ids')
-                if move in produce_moves and self.lot_id:
-                    moves_lines = moves_lines.filtered(lambda ml: self.lot_id in ml.produce_line_ids.lot_id)  # FIXME sle: double check with arm
-                for move_line in moves_lines:
-                    # Iterate over all move_lines until we unbuilded the correct quantity.
-                    taken_quantity = min(needed_quantity, move_line.qty_done - qty_already_used[move_line])
-                    if taken_quantity:
-                        move_line_vals = self._prepare_move_line_vals(move, move_line, taken_quantity)
-                        self.env["stock.move.line"].create(move_line_vals)
-                        needed_quantity -= taken_quantity
-                        qty_already_used[move_line] += taken_quantity
-            else:
+            original_move = move in produce_moves and self.mo_id.move_raw_ids or self.mo_id.move_finished_ids
+            original_move = original_move.filtered(lambda m: m.product_id == move.product_id)
+            if not original_move:
                 move.quantity_done = float_round(move.product_uom_qty, precision_rounding=move.product_uom.rounding)
+                continue
+            needed_quantity = move.product_uom_qty
+            moves_lines = original_move.mapped('move_line_ids')
+            if move in produce_moves and self.lot_id:
+                moves_lines = moves_lines.filtered(lambda ml: self.lot_id in ml.produce_line_ids.lot_id)  # FIXME sle: double check with arm
```

Comment:
This might be a good opportunity to actually fix this `FIXME` ;) Let me double check with ARM :P